Turn off clap with older cmakes

commit 5f265e

```
-- Creating CLAP surge-fx_CLAP from surge-fx
CMake Error at libs/clap-juce-extensions/CMakeLists.txt:51 (get_property):
  get_property could not find TARGET clap_juce_sources.  Perhaps it has not
  yet been created.
Call Stack (most recent call first):
  src/surge-fx/CMakeLists.txt:65 (clap_juce_extensions_plugin)


-- Creating CLAP surge-xt_CLAP from surge-xt
CMake Error at libs/clap-juce-extensions/CMakeLists.txt:51 (get_property):
  get_property could not find TARGET clap_juce_sources.  Perhaps it has not
  yet been created.
Call Stack (most recent call first):
  src/surge-xt/CMakeLists.txt:53 (clap_juce_extensions_plugin)


CMake Error at src/surge-xt/CMakeLists.txt:55 (target_sources):
  Cannot specify sources for target "surge-xt_CLAP" which is not built by
  this project.


-- Including JUCE VST3 host-side context menu support...
-- Including JUCE VST3 client extensions...
-- Validate me! Please!
-- Creating pluginval target for plugin: Surge XT.vst3
-- Creating pluginval target for plugin: .vst3
-- Adding download-extra-content target from https://github.com/surge-synthesizer/surge-extra-content.git
-- Configuring incomplete, errors occurred!
See also "/home/studio/dev/git/surge/build/CMakeFiles/CMakeOutput.log".
See also "/home/studio/dev/git/surge/build/CMakeFiles/CMakeError.log".
```
[CMakeError.log](https://github.com/surge-synthesizer/surge/files/8010553/CMakeError.log)
[CMakeOutput.log](https://github.com/surge-synthesizer/surge/files/8010554/CMakeOutput.log)

new clone, same error

```
-- Creating CLAP surge-fx_CLAP from surge-fx
CMake Error at libs/clap-juce-extensions/CMakeLists.txt:51 (get_property):
  get_property could not find TARGET clap_juce_sources.  Perhaps it has not
  yet been created.
Call Stack (most recent call first):
  src/surge-fx/CMakeLists.txt:65 (clap_juce_extensions_plugin)
```
etc...

apologies... here's the whole thing...

```
cmake -Bbuild                          
-- It's Surge XT, folks! Version is 1.1.0
-- CMAKE_BUILD_TYPE is Release
-- CMake version is 3.18.4
-- Compiler version is 10.2.1
-- Targeting 64-bit configuration
-- Setting -no-pie on EXE flags; use SURGE_SKIP_PIE_CHANGE=TRUE to avoid
-- Holding off on AVX support. See #4479 for the strategy
-- Using JUCE from /home/studio/dev/git/surge/libs/JUCE
-- Configuring juceaide
-- Building juceaide
-- Exporting juceaide
CMake Error at libs/clap-juce-extensions/clap-libs/clap-helpers/CMakeLists.txt:8 (set_target_properties):
  INTERFACE_LIBRARY targets may only have whitelisted properties.  The
  property "POSITION_INDEPENDENT_CODE" is not allowed.


CMake Error at libs/clap-juce-extensions/clap-libs/clap-helpers/CMakeLists.txt:8 (set_target_properties):
  INTERFACE_LIBRARY targets may only have whitelisted properties.  The
  property "CXX_STANDARD" is not allowed.


CMake Error at libs/clap-juce-extensions/CMakeLists.txt:30 (add_library):
  add_library INTERFACE library requires no source arguments.


CMake Error at libs/clap-juce-extensions/CMakeLists.txt:34 (target_include_directories):
  Cannot specify include directories for target "clap_juce_sources" which is
  not built by this project.


-- Building Surge XT using the following JUCE wrappers: VST3;Standalone
-- Module support is disabled.
-- Version: 8.0.1
-- Build type: Release
-- CXX_STANDARD: 17
-- Required features: cxx_variadic_templates
-- Checking processor clipping capabilities...
-- Checking processor clipping capabilities... none
-- Including local sqlite
-- Checking LJ_TARGET_X64 /home/studio/dev/git/surge/libs/LuaJitLib/LuaJIT;/home/studio/dev/git/surge/libs/LuaJitLib/LuaJIT/src;/home/studio/dev/git/surge/build/libs/LuaJitLib
-- LuaJIT target: x64
-- LuaJIT target: x64
-- filesystem: Using ghc::filesystem from "filesystem/ghc-filesystem.h"
-- Creating fs import via '/home/studio/dev/git/surge/build/libs/sst/sst-plugininfra/libs/filesystem/include/filesystem/import.h'
-- Creating CLAP surge-fx_CLAP from surge-fx
CMake Error at libs/clap-juce-extensions/CMakeLists.txt:51 (get_property):
  get_property could not find TARGET clap_juce_sources.  Perhaps it has not
  yet been created.
Call Stack (most recent call first):
  src/surge-fx/CMakeLists.txt:65 (clap_juce_extensions_plugin)


-- Creating CLAP surge-xt_CLAP from surge-xt
CMake Error at libs/clap-juce-extensions/CMakeLists.txt:51 (get_property):
  get_property could not find TARGET clap_juce_sources.  Perhaps it has not
  yet been created.
Call Stack (most recent call first):
  src/surge-xt/CMakeLists.txt:53 (clap_juce_extensions_plugin)


CMake Error at src/surge-xt/CMakeLists.txt:55 (target_sources):
  Cannot specify sources for target "surge-xt_CLAP" which is not built by
  this project.


-- Including JUCE VST3 host-side context menu support...
-- Including JUCE VST3 client extensions...
-- Validate me! Please!
-- Creating pluginval target for plugin: Surge XT.vst3
-- Creating pluginval target for plugin: .vst3
-- Adding download-extra-content target from https://github.com/surge-synthesizer/surge-extra-content.git
-- Configuring incomplete, errors occurred!
See also "/home/studio/dev/git/surge/build/CMakeFiles/CMakeOutput.log".
See also "/home/studio/dev/git/surge/build/CMakeFiles/CMakeError.log".
```

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -13,7 +13,19 @@ include(cmake/lib.cmake)
 
 message(STATUS "Using JUCE from ${SURGE_JUCE_PATH}")
 add_subdirectory(${SURGE_JUCE_PATH} ${CMAKE_BINARY_DIR}/JUCE EXCLUDE_FROM_ALL)
-add_subdirectory(../libs/clap-juce-extensions clap-extensions EXCLUDE_FROM_ALL)
+
+if(${CMAKE_VERSION} VERSION_LESS 3.21)
+  message(WARNING "CMake version less than 3.21. Skipping clap builds. Please "
+    "consider using a newer version of CMake")
+  set(SURGE_BUILD_CLAP FALSE)
+else()
+  message(STATUS "CMake version ${CMAKE_VERSION} allows CLAP build")
+  set(SURGE_BUILD_CLAP TRUE)
+endif()
+
+if(SURGE_BUILD_CLAP)
+  add_subdirectory(../libs/clap-juce-extensions clap-extensions EXCLUDE_FROM_ALL)
+endif()
 
 add_library(simde INTERFACE)
 target_include_directories(simde INTERFACE ${CMAKE_SOURCE_DIR}/libs/simde)

--- a/src/cmake/pluginval.cmake
+++ b/src/cmake/pluginval.cmake
@@ -1,24 +1,24 @@
   message(STATUS "Validate me! Please!")
 
-  if (APPLE)
+  if(APPLE)
     set(pluginval_url "https://github.com/Tracktion/pluginval/releases/latest/download/pluginval_macOS.zip")
     set(pluginval_exe pluginval/pluginval.app/Contents/MacOS/pluginval)
-  elseif (WIN32)
+  elseif(WIN32)
     set(pluginval_url "https://github.com/Tracktion/pluginval/releases/latest/download/pluginval_Windows.zip")
     set(pluginval_exe pluginval/pluginval.exe)
-  else ()
+  else()
     set(pluginval_url "https://github.com/Tracktion/pluginval/releases/latest/download/pluginval_Linux.zip")
     set(pluginval_exe pluginval/pluginval)
-  endif ()
+  endif()
 
   add_custom_target(stage-pluginval)
   add_custom_command(TARGET stage-pluginval
-          POST_BUILD
-          WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
-          COMMAND cmake -E make_directory pluginval
-          COMMAND curl -L ${pluginval_url} -o pluginval/pluginval.zip
-          COMMAND cd pluginval && unzip -o pluginval.zip
-          )
+    POST_BUILD
+    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+    COMMAND cmake -E make_directory pluginval
+    COMMAND curl -L ${pluginval_url} -o pluginval/pluginval.zip
+    COMMAND cd pluginval && unzip -o pluginval.zip
+    )
 
   add_custom_target(surge-pluginval-all)
   function(create_pluginval_target name target plugin)
@@ -28,17 +28,20 @@
     add_dependencies(${name} stage-pluginval)
     get_target_property(plugin_location ${target} LIBRARY_OUTPUT_DIRECTORY)
     add_custom_command(TARGET ${name}
-            POST_BUILD
-            WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
-            COMMAND ${pluginval_exe} --validate-in-process --output-dir "." --validate "${plugin_location}/${plugin_name}" || exit 1
-            )
+      POST_BUILD
+      WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+      COMMAND ${pluginval_exe} --validate-in-process --output-dir "." --validate "${plugin_location}/${plugin_name}" || exit 1
+      )
     add_dependencies(surge-pluginval-all ${name})
   endfunction()
 
-  create_pluginval_target(surge-xt-pluginval-vst3 surge-xt_VST3 "Surge XT.vst3")
-  create_pluginval_target(surge-fx-pluginval-vst3 surge-fx_VST3 "${SURGE_FX_PRODUCT_NAME}.vst3")
+  get_target_property(fxn surge-fx JUCE_PRODUCT_NAME)
+  get_target_property(xtn surge-xt JUCE_PRODUCT_NAME)
 
-  if (APPLE)
-    create_pluginval_target(surge-xt-pluginval-au surge-xt_AU "Surge XT.component")
-    create_pluginval_target(surge-fx-pluginval-au surge-fx_AU "${SURGE_FX_PRODUCT_NAME}.component")
-  endif ()
+  create_pluginval_target(surge-xt-pluginval-vst3 surge-xt_VST3 "${xtn}.vst3")
+  create_pluginval_target(surge-fx-pluginval-vst3 surge-fx_VST3 "${fxn}.vst3")
+
+  if(APPLE)
+    create_pluginval_target(surge-xt-pluginval-au surge-xt_AU "${xtn}.component")
+    create_pluginval_target(surge-fx-pluginval-au surge-fx_AU "${fxn}.component")
+  endif()

--- a/src/surge-fx/CMakeLists.txt
+++ b/src/surge-fx/CMakeLists.txt
@@ -10,11 +10,11 @@ endif()
 file(GLOB SURGE_FX_BANK_RESOURCES_GLOB
   ${CMAKE_SOURCE_DIR}/resources/surge-fx/*.svg
   ${CMAKE_SOURCE_DIR}/resources/surge-fx/icons/*.svg
-)
+  )
 
 juce_add_binary_data(surge-fx-binary
   SOURCES ${SURGE_FX_BANK_RESOURCES_GLOB}
-)
+  )
 
 juce_add_plugin(${PROJECT_NAME}
   PRODUCT_NAME ${SURGE_FX_PRODUCT_NAME}
@@ -44,7 +44,7 @@ juce_add_plugin(${PROJECT_NAME}
   LV2_SHARED_LIBRARY_NAME SurgeXT_FX
 
   FORMATS ${SURGE_JUCE_FORMATS}
-)
+  )
 
 target_sources(${PROJECT_NAME} PRIVATE
   SurgeFXEditor.cpp
@@ -52,7 +52,7 @@ target_sources(${PROJECT_NAME} PRIVATE
   SurgeFXProcessor.cpp
   SurgeFXProcessor.h
   SurgeLookAndFeel.h
-)
+  )
 
 target_link_libraries(${PROJECT_NAME} PRIVATE
   surge::surge-common
@@ -60,9 +60,11 @@ target_link_libraries(${PROJECT_NAME} PRIVATE
   surge-juce
   juce::juce_audio_utils
   juce::juce_audio_processors
-)
+  )
 
-clap_juce_extensions_plugin(TARGET surge-fx
-        CLAP_ID "org.surge-synth-team.surge-xt-fx")
+if(SURGE_BUILD_CLAP)
+  clap_juce_extensions_plugin(TARGET surge-fx
+    CLAP_ID "org.surge-synth-team.surge-xt-fx")
+endif()
 
 surge_juce_package(${PROJECT_NAME} "Surge XT Effects")

--- a/src/surge-xt/CMakeLists.txt
+++ b/src/surge-xt/CMakeLists.txt
@@ -50,9 +50,11 @@ juce_add_plugin(${PROJECT_NAME}
   FORMATS ${SURGE_JUCE_FORMATS}
   )
 
-clap_juce_extensions_plugin(TARGET surge-xt
-        CLAP_ID "org.surge-synth-team.surge-xt")
-target_sources(${PROJECT_NAME}_CLAP PRIVATE plugin_type_extensions/SurgeSynthClapExtensions.cpp)
+if(SURGE_BUILD_CLAP)
+  clap_juce_extensions_plugin(TARGET surge-xt
+    CLAP_ID "org.surge-synth-team.surge-xt")
+  target_sources(${PROJECT_NAME}_CLAP PRIVATE plugin_type_extensions/SurgeSynthClapExtensions.cpp)
+endif()
 
 if(JUCE_ASIO_SUPPORT)
   target_compile_definitions(${PROJECT_NAME} PUBLIC JUCE_ASIO=1)

